### DISCUSSION TITLE
What changes are required to annotation processors with 3.6.x?

### DISCUSSION THREAD

**Discussion by ronishz** [micronaut-projects/micronaut-core#7898] [Q&A] [answered]
I'm getting below error when compiling code with 3.6.0 and 3.6.1:

void io.micronaut.annotation.processing.visitor.JavaClassElement.<init>(javax.lang.model.element.TypeElement)

AnnotationProcessorsPath in my project's pom has micronaut-inject-java

Happy to hear on resolutions! Thanks!

**Comment by yawkat** [comment]
can you show the full error please?

**Reply by ronishz** [reply]
Unfortunately ahead of this line in error it only prints other class names without any actual cause.

**Reply by ronishz** [reply]
I've micronaut-inject-java dependency in my pom under annotationProcessors.  Do I need to change it to something else with 3.6.x?

**Comment by ronishz** [comment]
![16609298797664321760263652848332](https://user-images.githubusercontent.com/17245184/185674243-c8649a29-0f84-47b0-91c2-f4b3d47b9762.jpg)
Here's the complete error

**Reply by ronishz** [reply]
It compiles when I remove micronaut-inject-java dependency but again fails when I run maven tests as it fails to identify classes (annotationProcessing fails)

**Reply by ronishz** [reply]
To add, I can now see java.lang.NoSuchMethodError before the above error posted in picture

**Comment by ronishz** [comment, marked as answer]
Hello, here's an update.

The root cause was conflict of annotation Processors. Found out that micronaut-openapi was conflicting with Micronaut-inject-java. Removal of openapi from AnnotationProcessorsPath fixed the issue. Thanks!
